### ISSUE TITLE
flutter 3.24.1 create test drive run failed on android

### ISSUE THREAD

**Issue by imatespl** [flutter/flutter#153989]
### Steps to reproduce

1. install flutter sdk
2. in windows cmd run `flutter create test_drive`
3. then run `flutter run`, give this error
```
D:\test\test1>flutter run
Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!
Launching lib\main.dart on EDI AL10 in debug mode...

FAILURE: Build failed with an exception.

* Where:
Build file 'D:\test\test1\android\app\build.gradle' line: 3

* What went wrong:
Plugin [id: 'kotlin-android'] was not found in any of the following sources:

- Gradle Core Plugins (not a core plugin, please see https://docs.gradle.org/7.6.3/userguide/standard_plugins.html for available core plugins)
- Included Builds (None of the included builds contain this plugin)
- Plugin Repositories (plugin dependency must include a version number for this source)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 16s
Running Gradle task 'assembleDebug'...                             17.1s
Error: Gradle task assembleDebug failed with exit code 1
```
4. I change the `kotlin-android` plugin version  from 1.7.10 to 2.0.20 in androdi/settings.gradle, this error can fixed, but give the other error
```
plugins {
    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
    id "com.android.application" version "7.3.0" apply false
    id "org.jetbrains.kotlin.android" version "2.0.20" apply false
}
```
the error is:
```
* what went wrong: an exception occurred applying plugin request [id: 'dev.flutter.flutter-gradle-plugin'] 
failed to apply plugin 'dev.flutter.flutter-gradle-plugin'. 
could not get unknown property 'usesunsupporteddependencyversions' for project ':app' of type org.gradle.api.project.
```

### Expected results

can build the android app and debug

### Actual results

can not build the android app

### Code sample

this code is the flutter.dev doc, create the test_drive example code

**Comment by imatespl** [comment]
i change the flutter 3.24.1 to 3.22.3 and change kotlin-android plugin to 2.0.20, can build the android app

**Comment by imatespl** [comment]
this diff of build.gradle in android/app
![Uploading diff_of_build.gradle.png…]()

**Comment by huycozy** [comment]
Hi @imatespl 
Please provide the output of `flutter doctor -v`. This could be similar to https://github.com/flutter/flutter/issues/146281 which is about gradle issue. 

Can you check this https://github.com/flutter/flutter/issues/151044#issuecomment-2206052979 to see if it works for you? If not, can you please share the sample project here so that I can try it on my end?

**Comment by imatespl** [comment]
This the work good version in android studio:
PS D:\test_drive> flutter doctor -v
Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source! 
[√] Flutter (Channel stable, 3.22.3, on Microsoft Windows [版本 10.0.17763.6189], locale zh-CN) 
    • Flutter version 3.22.3 on channel stable at C:\Users\lenovo\AppData\Local\flutter-sdk\flutter 
    • Upstream repository https://github.com/flutter/flutter.git                                    
    • Framework revision b0850beeb2 (7 weeks ago), 2024-07-16 21:43:41 -0700                        
    • Engine revision 235db911ba
    • Dart version 3.4.4
    • DevTools version 2.34.3
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[√] Windows Version (Installed version of Windows is version 10 or higher) 

[√] Android toolchain - develop for Android devices (Android SDK version 35.0.0) 
    • Android SDK at C:\Users\lenovo\AppData\Local\Android\Sdk 
    • Platform android-35, build-tools 35.0.0
    • ANDROID_HOME = C:\Users\lenovo\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11572160)
    • All Android licenses accepted.

[√] Chrome - develop for the web 
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop Windows apps 
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components 

[√] Android Studio (version 2024.1) 
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from: 
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11572160)

[√] IntelliJ IDEA Community Edition (version 2023.1) 
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2023.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code (version 1.93.0) 
    • VS Code at C:\Users\lenovo\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.96.0

[√] Connected device (4 available) 
    • EDI AL10 (mobile) • 8DF6R17112006444 • android-arm64  • Android 8.0.0 (API 26)
    • Windows (desktop) • windows          • windows-x64    • Microsoft Windows [版本 10.0.17763.6189]
    • Chrome (web)      • chrome           • web-javascript • Google Chrome 128.0.6613.119
    • Edge (web)        • edge             • web-javascript • Microsoft Edge 128.0.2739.42

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
PS D:\test_drive>

**Comment by imatespl** [comment]
> Hi @imatespl Please provide the output of `flutter doctor -v`. This could be similar to #146281 which is about gradle issue.
> 
> Can you check this [#151044 (comment)](https://github.com/flutter/flutter/issues/151044#issuecomment-2206052979) to see if it works for you? If not, can you please share the sample project here so that I can try it on my end?

I will try  [#151044 (comment)](https://github.com/flutter/flutter/issues/151044#issuecomment-2206052979)

**Comment by imatespl** [comment]
The  [#151044 (comment)](https://github.com/flutter/flutter/issues/151044#issuecomment-2206052979) upgrade the gradle version, from 6.7.1 to 7.6.3, but the new flutter version create the app already use the 7.6.3

**Comment by huycozy** [comment]
Can you try these versions instead?

```
plugins {
    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
    id "com.android.application" version "8.1.1" apply false
    id "org.jetbrains.kotlin.android" version "1.7.10" apply false
}
```

```
distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-all.zip
```

**Comment by imatespl** [comment]
I install the flutter 3.24.1, create a project ,the gradle is 7.6.3, the bug is not fixed, i install the 3.24.2, create a project, the gradle is 8.3, the bug is fixed, close the issues

**Comment by huycozy** [comment]
Yeah, I think it's due to the template with newer Gradle version has been landed recently by https://github.com/flutter/flutter/issues/153885.
